packages/protocol/scripts/build.ts

```diff
@@ -0,0 +1,143 @@
+/* tslint:disable no-console */
+import { execSync } from 'child_process'
+import * as fs from 'fs'
+import * as path from 'path'
+
+const ROOT_DIR = path.normalize(path.join(__dirname, '../'))
+const BUILD_DIR = path.join(ROOT_DIR, 'build')
+const CONTRACTKIT_GEN_DIR = path.normalize(path.join(ROOT_DIR, '../contractkit/src/generated'))
+
+export const ProxyContracts = [
+  'GasCurrencyWhitelistProxy',
+  'GasPriceMinimumProxy',
+  'MultiSigProxy',
+  'BondedDepositsProxy',
+  'AttestationsProxy',
+  'EscrowProxy',
+  'ExchangeProxy',
+  'GoldTokenProxy',
+  'ReserveProxy',
+  'StableTokenProxy',
+  'SortedOraclesProxy',
+  'GovernanceProxy',
+  'RegistryProxy',
+]
+export const CoreContracts = [
+  // common
+
+  'GasPriceMinimum',
+  'GasCurrencyWhitelist',
+  'MultiSig',
+  'Registry',
+  'Validators',
+
+  // governance
+  'BondedDeposits',
+  'Governance',
+
+  // identity
+  'Attestations',
+  'Escrow',
+  'Random',
+
+  // stability
+  'Exchange',
+  'GoldToken',
+  'Reserve',
+  'StableToken',
+  'SortedOracles',
+]
+
+const OtherContracts = [
+  'Proxy',
+  'Migrations',
+  // abstract
+  'Initializable',
+  'UsingRegistry',
+]
+export const ImplContracts = OtherContracts.concat(ProxyContracts).concat(CoreContracts)
+
+// const TruffleTestContracts = ['Ownable'].concat(OtherContracts).concat(CoreContracts)
+
+function getArtifact(contractName: string) {
+  const file = fs.readFileSync(`${BUILD_DIR}/contracts/${contractName}.json`).toString()
+  return JSON.parse(file)
+}
+
+function exec(cmd: string) {
+  return execSync(cmd, { cwd: ROOT_DIR, stdio: 'inherit' })
+}
+
+function hasEmptyBytecode(contract: any) {
+  return contract.bytecode === '0x'
+}
+
+function compile() {
+  console.log('Compiling')
+
+  exec(`yarn run --silent truffle compile --build_directory=${BUILD_DIR}`)
+
+  for (const contractName of ImplContracts) {
+    const fileStr = getArtifact(contractName)
+    if (hasEmptyBytecode(fileStr)) {
+      console.error(
+        `${contractName} has empty bytecode. Maybe you forgot to fully implement an interface?`
+      )
+      process.exit(1)
+    }
+  }
+}
+
+function generateFilesForTruffle() {
+  console.log('protocol: Generating Truffle Types')
+  exec(`rm -rf "${ROOT_DIR}/typechain"`)
+
+  // const globPattern = `${BUILD_DIR}/contracts/@(${TruffleTestContracts.join('|')}).json`
+  const globPattern = `${BUILD_DIR}/contracts/*.json`
+  exec(
+    `yarn run --silent typechain --target=truffle --outDir "./types/typechain" "${globPattern}" `
+  )
+}
+
+function generateFilesForContractKit() {
```

Comment:
I would love for this functionality to be available in contractkit so that external developers can use this for their contracts!